packages/manager/src/queries/events.ts

```diff
@@ -0,0 +1,133 @@
+import { APIError, ResourcePage } from '@linode/api-v4/lib/types';
+import {
+  useQuery,
+  useInfiniteQuery,
+  useQueryClient,
+  InfiniteData,
+} from 'react-query';
+import {
+  Entity,
+  Event,
+  getEvents,
+  markEventSeen,
+} from '@linode/api-v4/lib/account';
+import { isInProgressEvent } from 'src/store/events/event.helpers';
+import { isNotNullOrUndefined } from 'src/utilities/nullOrUndefined';
+
+export interface ExtendedEvent extends Event {
+  _deleted?: string;
+  _initial?: boolean;
+}
+
+export interface EntityEvent extends Omit<Event, 'entity'> {
+  entity: Entity;
+}
+
+const queryKey = 'events';
+
+export const useEventsPolling = (
+  eventHandler?: (event: ExtendedEvent) => void
+) => {
+  useQuery<ExtendedEvent[], APIError[]>(
```

Comment:
Can we avoid using extended types?

```suggestion
  useQuery<Event[], APIError[]>(
```